### PR TITLE
web/storagenode: minimal allowed version view implemented

### PR DESCRIPTION
**Pull request by VitaliiShpital**
What: 
now user can check current minimal allowed version

Why:
SNO Dashboard enhancements

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: private/version/checker/service.go [hunk 1]
```diff
@@ -85,31 +86,42 @@ func (srv *Service) Run(ctx context.Context) (err error) {
...
 	}
 	srv.mu.Lock()
 	defer srv.mu.Unlock()
-	return srv.allowed
+	return srv.acceptedVersion, srv.allowed
 }
 
 // CheckVersion checks if the client is running latest/allowed code
 func (srv *Service) checkVersion(ctx context.Context) (allowed bool) {
 	defer mon.Task()(&ctx)(nil)
```
Comment (crawter) [inline private/version/checker/service.go]: var err error
defer mon.Task()(&ctx)(&err)

File: private/version/checker/service.go [hunk 2]
```diff
@@ -85,31 +86,42 @@ func (srv *Service) Run(ctx context.Context) (err error) {
...
-		return false
+		return version.OldSemVer{}, false
 	}
 	srv.mu.Lock()
 	defer srv.mu.Unlock()
-	return srv.allowed
+	return srv.acceptedVersion, srv.allowed
 }
 
 // CheckVersion checks if the client is running latest/allowed code
```
Comment (crawter) [inline private/version/checker/service.go]: CheckVersion --> checkVersion

File: web/storagenode/src/storagenode/dashboard.ts [hunk 3]
```diff
@@ -53,3 +54,21 @@ export class BandwidthInfo {
...
+    public constructor(
+        public major: number,
+        public minor: number,
+        public patch: number) {}
+
+    /**
+     * Converts allowed version numbers to string type
+     * @returns allowed version - string of allowed version value
+     */
+    public toString(): string {
```
Comment (stefanbenten) [inline web/storagenode/src/storagenode/dashboard.ts]: normally those functions are just called String()

File: private/version/checker/service.go [hunk 4]
```diff
@@ -85,31 +86,42 @@ func (srv *Service) Run(ctx context.Context) (err error) {
 }
 
 // IsAllowed returns whether if the Service is allowed to operate or not
-func (srv *Service) IsAllowed(ctx context.Context) bool {
+func (srv *Service) IsAllowed(ctx context.Context) (version.OldSemVer, bool) {
```
Comment (stefanbenten) [inline private/version/checker/service.go]: We should be using the new version struct, as the old should not be used going forward.

File: private/version/checker/service.go [hunk 5]
```diff
@@ -85,31 +86,42 @@ func (srv *Service) Run(ctx context.Context) (err error) {
...
 }
 
-// CheckVersion checks if the client is running latest/allowed code
+// checkVersion checks if the client is running latest/allowed code
 func (srv *Service) checkVersion(ctx context.Context) (allowed bool) {
-	defer mon.Task()(&ctx)(nil)
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	var minimum version.OldSemVer
```
Comment (stefanbenten) [inline private/version/checker/service.go]: Same as comment above, we should use the new version struct

File: web/storagenode/src/storagenode/dashboard.ts (hunk 3 above)
Comment (VitaliiShpital) [reply web/storagenode/src/storagenode/dashboard.ts]: but not in js, java or c#

File: storagenode/console/service.go [hunk 6]
```diff
@@ -120,8 +120,9 @@ type Dashboard struct {
 	LastPingFromID      storj.NodeID `json:"lastPingFromID"`
 	LastPingFromAddress string       `json:"lastPingFromAddress"`
 
-	Version  version.SemVer `json:"version"`
-	UpToDate bool           `json:"upToDate"`
+	Version        version.SemVer    `json:"version"`
+	AllowedVersion version.OldSemVer `json:"allowedVersion"`
```
Comment (stefanbenten) [inline storagenode/console/service.go]: This should be a new SemVer

File: storagenode/console/service.go [hunk 7]
```diff
@@ -134,10 +135,10 @@ func (s *Service) GetDashboardData(ctx context.Context) (_ *Dashboard, err error
 	data.NodeID = s.contact.Local().Id
 	data.Wallet = s.walletAddress
 	data.Version = s.versionInfo.Version
-	data.UpToDate = s.version.IsAllowed(ctx)
 	data.StartedAt = s.startedAt
 
 	data.LastPinged = s.pingStats.WhenLastPinged()
+	data.AllowedVersion, data.UpToDate = s.version.IsAllowed(ctx)
```
Comment (stefanbenten) [inline storagenode/console/service.go]: This should fail to build. You are assigning a new semver to the old struct.

File: web/storagenode/src/storagenode/dashboard.ts (hunk 3 above)
Comment (stefanbenten) [reply web/storagenode/src/storagenode/dashboard.ts]: This function needs to change afaik with the use of the new struct

File: storagenode/console/service.go (hunk 7 above)
Comment (VitaliiShpital) [reply storagenode/console/service.go]: You're right. Thanks for review

Comment (VitaliiShpital) [reply storagenode/console/service.go]: You're right. Thanks for review

**Pull request merged**